text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,337 @@
+2. Upgrading version of `aws-cdk` and other toolchain packages to version `2.x`.
+3. Rewriting import statements in source code if necessary, and how to use the [import rewriter]()
+4. Removing usage of deprecated APIs.
+  - List of APIs being deprecated and links to the suggested alternatives for each.
+5. Removing references to old feature flags
+  - List of feature flags being removed and strategy for migrating to new behavior for each.
+
+### Additional Candidates for Inclusion:
+
+The following are changes whose implementation's have been identified as requiring a breaking change. Inclusion in the v2 release is up for debate. Arguments to include a specific feature should take into account any increase in cost to the development team required to ship the release, and to users required to migrate their apps to the new version. Though we want to keep both of these costs as low as possible, a breaking release is something that happens relatively infrequently, so the opportunity should be taken to include features determined to have a high payoff to users.
+
+In addition, since the v1 modules will continue to be supported for some time, and the `aws-cdk-lib` package will be made of up these modules, the inclusion of any of these candidate features would need to be implemented in a way that didn't affect the v1 modules. How this would be done has not been determined, but may be affected by the nature of any particular change.
+
+#### Recommended
+
+The following are good candidates for inclusion in this major version release
+
+- [Easier Identification of Experimental Modules and APIs](https://github.com/aws/aws-cdk-rfcs/issues/116)
+
+There have been long running discussions about separating experimental APIs into a separate module to make them a more explicit "opt in" feature. Users would either have to install a different module (i.e. `aws-cdk-lib-exerimental`) or import from a different entrypoint (`import * as s3 from 'aws-cdk-lib/experimental`). Since this could be performed at build time and be an optional setting thats turned on or off, it could be introduced in the 2.0 release without affecting the v1 modules, this seems like an ideal time to implement. The technical details of how to achieve this are yet to be determined.
+
+- [Public S3 Artifacts](https://github.com/aws/aws-cdk-rfcs/issues/39)
+
+A significant downside of packaging all of the aws service construct libraries together is that the size of dependencies shipped to the users in most cases will be increased significantly. Being able to remove assets, like custom resource lambda code, that are currently bundled with the modules, could mitigate some of the impact on package size. This could be accomplished by packaging said assets, uploading them to S3, and then creating publically accessible lambda function layers referencing them across all AWS regions. Whether v2 of the cdk could accomplish this without impacting v1 has not been determined.
```

Comment:
It would be quite scary to me to have this maintenance burden of a set of public artifacts.

All of a sudden we'd be responsible for replicating these all across the world, have monitoring on them, have backups, have a release pipeline, ...

Reply:
Can we get some numbers?